packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx

```diff
@@ -14,130 +14,266 @@ import {
+    useDraftInboundLines(currentItem?.id);
 
-  useEffect(() => {
-    if (
-      wrappedInbound?.batches &&
-      Object.values(wrappedInbound?.batches).length === 0
-    ) {
-      onAddBatch();
-    }
-  });
-
-  return loading ? (
-    <BasicSpinner />
-  ) : (
-    <>
-      <InboundLineEditForm
-        draft={draft}
-        mode={mode}
-        item={item}
-        onChangeItem={onChangeItem}
-      />
-      <Divider margin={5} />
-      {wrappedInbound ? (
-        <TabContext value={currentTab}>
-          <Box flex={1} display="flex" justifyContent="space-between">
-            <Box flex={1} />
-            <Box flex={1}>
-              <TabList
-                value={currentTab}
-                centered
-                onChange={(_, v) => setCurrentTab(v)}
-              >
-                <Tab value={Tabs.Batch} label={Tabs.Batch} />
-                <Tab value={Tabs.Pricing} label={Tabs.Pricing} />
-              </TabList>
-            </Box>
-            <Box flex={1} justifyContent="flex-end" display="flex">
-              <ButtonWithIcon
-                color="primary"
-                variant="outlined"
-                onClick={onAddBatch}
-                label={t('label.add-batch')}
-                Icon={<PlusCircleIcon />}
-              />
-            </Box>
-          </Box>
-
-          <TableContainer
-            sx={{
-              height: isMediumScreen ? 300 : 400,
-              marginTop: 2,
-              borderWidth: 1,
-              borderStyle: 'solid',
-              borderColor: 'divider',
-              borderRadius: '20px',
-            }}
-          >
-            <Box
-              sx={{
-                width: 400,
-                height: isMediumScreen ? 300 : 400,
-                backgroundColor: theme =>
-                  alpha(theme.palette['background']['menu'], 0.4),
-                position: 'absolute',
-                borderRadius: '20px',
-              }}
-            />
-            <StyledTabPanel value={Tabs.Batch}>
-              <BatchTable batches={batches} />
-            </StyledTabPanel>
-
-            <StyledTabPanel value={Tabs.Pricing}>
-              <PricingTable batches={batches} />
-            </StyledTabPanel>
-          </TableContainer>
-        </TabContext>
+  return (
+    <Modal
+      title={
+        mode === ModalMode.Create
+          ? t('heading.add-item')
+          : t('heading.edit-item')
+      }
+      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
+      nextButton={
+        <DialogButton
+          variant="next"
+          onClick={async () => {
+            try {
+              await saveLines();
+              setCurrentItem(mode === ModalMode.Update ? nextItem : null);
+              return true;
+            } catch (e) {
+              return false;
+            }
+          }}
+        />
+      }
+      okButton={<DialogButton variant="ok" onClick={saveLines} />}
```

Comment:
The ok behaviour isn't what I expect - is this just me?
Currently, OK saves the lines. I would expect it to save and close the modal. If the button only saves, I expect it to say `Save`.
Also - there is a cancel button which is used to close, which means if I want to save and then close, I have to click `Ok` and then `Cancel` and in my mind, the cancel is used to invalidate something, so that process flow feels wrong.

I'd prefer the `Ok` to save and close. and the `Cancel` to abandon changes and close.
Otherwise, the `Ok` should say `Save` and `Cancel` should say `Close` - but that might be confusing with the detail view which saves without clicking save and so `Close` has a different meaning there.. or would, if we had one!

Reply:
Ah yes, sorry I missed this behaviour. Should be good in the upcoming PRs, I think!